pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl

```diff
@@ -0,0 +1,329 @@
+task rnaseqc2 {
+	input {
+		File bam_file
+		File genes_gtf
+		String sample_id
+		File exon_bed = "gs://gtex-resources/GENCODE/gencode.v26.GRCh38.insert_size_intervals_geq1000bp.bed"
+	}
+	
+	Int disk_space = ceil(size(bam_file, 'GB') + size(genes_gtf, 'GB')) + 100
+
+	command {
+		set -euo pipefail
+		echo $(date +"[%b %d %H:%M:%S] Running RNA-SeQC 2")
+		rnaseqc ~{genes_gtf} ~{bam_file} . -s ~{sample_id} -v --bed ~{exon_bed}
+		echo "  * compressing outputs"
+		gzip *.gct
+		echo $(date +"[%b %d %H:%M:%S] done")
+	}
+
+	output {
+		File gene_tpm = "${sample_id}.gene_tpm.gct.gz"
+		File gene_counts = "${sample_id}.gene_reads.gct.gz"
+		File exon_counts = "${sample_id}.exon_reads.gct.gz"
+		File insert_size_histogram = "${sample_id}.fragmentSizes.txt"
+		File metrics = "${sample_id}.metrics.tsv"
+	}
+
+	runtime {
+		docker: "us.gcr.io/broad-dsde-methods/ckachulis/rnaseqc:2.4.2"
+		memory: "10GB"
+		disks: "local-disk " + disk_space + " HDD"
+		preemptible: 0
+	}
+}
+
+task GetSampleName {
+	input {
+		File bam
+	}
+
+	parameter_meta {
+		bam : {
+			localization_optional : true
+		}
+	}
+
+	command <<<
+		gatk GetSampleName -I ~{bam} -O sample_name.txt
+	>>>
+
+	runtime {
+		docker: "us.gcr.io/broad-gatk/gatk:4.2.0.0"
+		disks: "local-disk 100 HDD"
+	}
+
+	output {
+		String sample_name = read_string("sample_name.txt")
+	}
+}
+
+task CopyReadGroupsToHeader {
+	input {
+		File bam_with_readgroups
+		File bam_without_readgroups
+	}
+
+	Int disk_size = ceil(2.0 * size([bam_with_readgroups, bam_without_readgroups], "GB")) + 10
+	String basename = basename(bam_without_readgroups)
+
+	command <<<
+		samtools view -H ~{bam_without_readgroups} > header.sam
+		samtools view -H ~{bam_with_readgroups} | grep "@RG" >> header.sam
+		samtools reheader header.sam ~{bam_without_readgroups} > ~{basename}
+	>>>
+
+	runtime {
+		docker: "us.gcr.io/broad-dsde-methods/samtools@sha256:0e49b0a5d91c203b8c07f5242277c2060b4b8ea54df8b1d123f990a1ad0588b2"
+		disks: "local-disk " + disk_size + " HDD"
+	}
+
+	output {
+		File output_bam = basename
+	}
+}
+
+task CollectRNASeqMetrics {
+	input {
+		File input_bam
+		File input_bam_index
+		String output_bam_prefix
+		File ref_dict
+		File ref_fasta
+		File ref_fasta_index
+		Int preemptible_tries
+		File ref_flat
+		File ribosomal_intervals
+	}
+
+	Float ref_size = size(ref_fasta, "GiB") + size(ref_fasta_index, "GiB") + size(ref_dict, "GiB")
+	Int disk_size = ceil(size(input_bam, "GiB") + ref_size) + 20
+
+	# This jar skips the header check of the ribosomal interval
+	File picard_jar = "gs://broad-dsde-methods-takuto/hydro.gen/picard_ignore_ribosomal_header.jar"
+
+	command {
+		java -Xms5000m -jar ~{picard_jar} CollectRnaSeqMetrics \
+		REF_FLAT=~{ref_flat} \
+		RIBOSOMAL_INTERVALS= ~{ribosomal_intervals} \
+		STRAND_SPECIFICITY=SECOND_READ_TRANSCRIPTION_STRAND \
+		INPUT=~{input_bam} \
+		OUTPUT=~{output_bam_prefix}.rna_metrics
+	}
+
+	runtime {
+		docker: "us.gcr.io/broad-gotc-prod/genomes-in-the-cloud:2.4.3-1564508330"
```

Comment:
Not even using the docker image here - Would be best to use an official release of Picard.